lib/std/os/uefi/protocol/serial_io.zig

```diff
@@ -0,0 +1,72 @@
+const std = @import("std");
+const uefi = std.os.uefi;
+const Guid = uefi.Guid;
+const Status = uefi.Status;
+const cc = uefi.cc;
+
+pub const SerialIo = extern struct {
+    revision: u64,
+    _reset: *const fn (*const SerialIo) callconv(cc) Status,
+    _set_attribute: *const fn (*const SerialIo, u64, u32, u32, ParityType, u8, StopBitsType) callconv(cc) Status,
+    _set_control: *const fn (*const SerialIo, u32) callconv(cc) Status,
+    _get_control: *const fn (*const SerialIo, *u32) callconv(cc) Status,
+    _write: *const fn (*const SerialIo, *usize, *anyopaque) callconv(cc) Status,
+    _read: *const fn (*const SerialIo, *usize, *anyopaque) callconv(cc) Status,
+
```

Comment:
Why aren't the `mode` and `device_type_guid` fields here?

Reply:
That was a miss on my part, they should be; I have added them now.

  A brief footnote on `device_type_guid`: in the spec it is described as a pointer that may be NULL though the typedef in the spec does not explicitly label it as Optional. I have made it optional as this mirrors what I see in the EDID Protocol in the standard library today. Let me know if I have made the wrong call here and I will revert it.